### ISSUE TITLE
Apollo Documentation

### ISSUE THREAD

**Issue by jpshelley** [guymers/babel-plugin-flow-relay-query#20]
I'm fairly new to flow and apollo, so I was hoping for improved documentation if possible. An example project would go a long way.
For example, are we required to have our schema.json referenced locally via:

```
const schema = require("??/schema.json");
```

I'm looking at adding this to a React Native project so would that mean having the schema downloaded locally?

**Comment by guymers** [comment]
Yes, this plugin uses the schema to check that the Flow definition corresponds to the what is in the schema.

```
Errors for fragment Article:
Expected type 'string', actual type 'number' for path author.email
Expected type 'string', actual type '?string' for path author.name
Expected type 'boolean', actual type 'number' for path sponsored
Expected type 'string', actual type '?string' for path title
```

This plugin works the same way as the Relay Babel plugin, add that section into a JavaScript file and reference it from Babel as a plugin.

```
{
  "presets": [
    "react-app"
  ],
  "plugins": [
    "./path/to/file"
  ]
}
```
The Relay examples do it a bit differently: https://github.com/relayjs/relay-examples/blob/master/relay-treasurehunt/.babelrc. I personally load Babel configuration as a JavaScript file.

I will try to find some time to write a basic example

**Comment by jpshelley** [comment]
Thanks for the quick reply, I'll try and give it a try again with the provided information. I know Apollo is working on flow code generation using https://github.com/apollographql/apollo-codegen so it will be good to see how each project differs and how they can benefit from each other.
